graphql/query_resolver.go

```diff
@@ -1033,8 +1033,18 @@ func (r *queryResolver) Waterfall(ctx context.Context, options WaterfallOptions)
 		// If no order options were specified, we're on the first page and should not put a limit on the first version returned so that we don't omit inactive versions
 		maxVersionOrder = 0
 	} else if maxOrderOpt == 0 {
-		// If we're paginating backwards, use the newest active version as the upper bound
-		maxVersionOrder = activeVersions[0].RevisionOrderNumber
+		// Find the next active version. If it doesn't exist, we're on the first page. If it does,
+		// set the max order to one less than its order so that we fetch all of the inactive versions
+		// in between.
+		nextActiveVersion, err := model.GetNextActiveWaterfallVersion(ctx, projectId, activeVersions[0].RevisionOrderNumber)
```

Comment:
Since we are aiming to check if the 0th version is actually the most recent waterfall version, might we instead use `GetMostRecentWaterfallVersion` and compare their IDs/hashes? It seems preferable to use an existing and hopefully optimized method.

Reply:
I'm not entirely sure if it's possible, if you imagine a scenario where this is the first page:

<img width="551" alt="Screenshot 2024-10-30 at 10 31 22 AM" src="https://github.com/user-attachments/assets/314f51c1-ea1a-430c-94bb-5fdc3d235c84">

If the most recent inactive commit is order 120, and Mohamed's first commit (0th `activeVersion`) is order 100, I don't know if we can reliably tell that we're on the first page. I think we'd somehow have to know that all of those versions are collapsed 😥

Because this method forces fetching all leading inactive versions, it also seems to help guarantee a consistent order so that things like this don't happen:

https://github.com/user-attachments/assets/7b279f7b-c4b6-4312-ba98-b9cd3a5833f4